collectors/modality-probe-collector-common/src/lib.rs

```diff
@@ -37,78 +42,52 @@ newtype! {
 }
 
 newtype! {
-    /// A log segment
-    ///
-    /// The log is divided into segments, each of which begins with some logical
-    /// clock entries and ends with a sequence of events. The id must be unique
-    /// within the session.
+    /// A log report sequence number
     #[derive(Debug, Eq, PartialEq, Hash, Copy, Clone)]
-    pub struct SegmentId(pub u32);
-}
-
-/// Map an event id to its name and description
-#[derive(Debug, Eq, PartialEq)]
-pub struct EventMapping {
-    id: EventId,
-    name: String,
-    description: String,
-}
-
-/// Map a probe id to its name and description
-#[derive(Debug, Eq, PartialEq)]
-pub struct ProbeMapping {
-    id: ProbeId,
-    name: String,
-    description: String,
+    pub struct SequenceNumber(pub u64);
 }
 
-/// The data that may be attached to a log entry
-#[derive(Debug, Eq, PartialEq, Clone)]
-pub enum LogEntryData {
-    Event(EventId),
-    EventWithPayload(EventId, u32),
-    LogicalClock(ProbeId, ProbeEpoch, ProbeTicks),
-}
+#[derive(Debug)]
+pub struct SerializationError;
 
-impl From<EventId> for LogEntryData {
-    fn from(val: EventId) -> LogEntryData {
-        LogEntryData::Event(val)
+impl From<ReportWireError> for SerializationError {
+    fn from(_: ReportWireError) -> Self {
+        SerializationError
     }
 }
 
-impl From<(ProbeId, ProbeEpoch, ProbeTicks)> for LogEntryData {
-    fn from((id, epoch, clock): (ProbeId, ProbeEpoch, ProbeTicks)) -> LogEntryData {
-        LogEntryData::LogicalClock(id, epoch, clock)
-    }
+#[derive(Debug, PartialEq)]
+pub struct Report {
+    pub probe_id: ProbeId,
+    pub probe_clock: LogicalClock,
+    pub seq_num: u64,
```

Comment:
We have a `SequenceNumber` type hanging out nearby, should we be using it here?

Reply:
c0adc77eab6c64507ba5d6b58460c064b1aa671d